src/Tools/ExternalAccess/Razor/RazorDynamicFileInfoProviderWrapper.cs

```diff
@@ -42,7 +42,8 @@ internal sealed class RazorDynamicFileInfoProviderWrapper : IDynamicFileInfoProv
             var result = await _innerDynamicFileInfoProvider.Value.GetDynamicFileInfoAsync(projectId, projectFilePath, filePath, cancellationToken).ConfigureAwait(false);
             var serviceProvider = new RazorDocumentServiceProviderWrapper(result.DocumentServiceProvider);
             var razorDocumentPropertiesService = result.DocumentServiceProvider.GetService<IRazorDocumentPropertiesService>();
-            var dynamicFileInfo = new DynamicFileInfo(result.FilePath, result.SourceCodeKind, result.TextLoader, razorDocumentPropertiesService.DesignTimeOnly, serviceProvider);
+            var designTimeOnly = razorDocumentPropertiesService?.DesignTimeOnly ?? false;
```

Comment:
I couldn't think of a better time to `#nullable enable` this file :)

Reply:
That was my first action last night, but sadly for me this whole project is already enabled.

Reply:
Hmm interesting, wonder why there wasn't a warning then - maybe DocumentServiceProvider isn't null annotated?

Reply:
I guess `IRazorDocumentServiceProvider.GetService` is incorrectly annotated to be non-nullable return. How big a deal is it to change the nullability on an interface in the external access layer?

Reply:
I think it's a big deal - @jasonmalinowski was changing the null-ability of a method a breaking change?

Reply:
It's never a binary breaking change. If the method was simply mis-annotated then we should change it, and it may be a source breaking change for anybody who referenced that and is now getting build warnings (which may become build errors if configured appropriately). That said that's the whole point of the feature, so we don't shoot the messenger. 😄 

Obviously if you're changing an API to start returning null where it wasn't before, that's just potentially a breaking change in the sense of changing API expectations, but in that case it's still the right thing to reannotate.

Reply:
It looks like this is unannotated:

https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentServiceProvider.cs

My rule for fixing null refs (doubly so when we had to roll it back) is keep annotating until the compiler finds the bug, and then fix the bug. That way we prevent the mistake again.

Reply:
This wasn't IDocumentServiceProvider, but I take your point about annotating all the things, so I'll do it in a follow up. The build finally went green after the spanish test flakiness so no point making EnC wait.